notification/events.go

```diff
@@ -231,9 +231,10 @@ func addNotificationEvent(ctx context.Context, id string, celEnv map[string]any,
 			continue
 		}
 
-		newEvent := api.Event{
+		newEvent := models.Event{
 			Name:       api.EventNotificationSend,
 			Properties: payload.AsMap(),
+			Delay:      n.WaitFor,
```

Comment:
I think we still need to insert a pending notification into notification send.history and then update it on final send